lib/WP_Auth0_Routes.php

```diff
@@ -44,10 +49,35 @@ public function custom_requests( $wp ) {
 			case 'oauth2-config': $this->oauth2_config(); exit;
 			case 'migration-ws-login': $this->migration_ws_login(); exit;
 			case 'migration-ws-get-user': $this->migration_ws_get_user(); exit;
+			case 'coo-fallback': $this->coo_fallback(); exit;
 			}
 		}
 	}
 
+	protected function coo_fallback() {
+		$cdn = $this->a0_options->get( 'auth0js-cdn' );
+		$client_id = $this->a0_options->get( 'client_id' );
+		$domain = $this->a0_options->get( 'domain' );
+		$redirect_uri = home_url( '/index.php?auth0=1', $this->a0_options->get( 'force_https_callback' ) );
+		echo <<<EOT
+		<!DOCTYPE html>
+		<html>
+		<head>
+		<script src="$cdn"></script>
+		<script type="text/javascript">
+		  var auth0 = new auth0.WebAuth({
+			clientID: '$client_id',
+			domain: '$domain',
+			redirectUri: '$redirect_uri'
+		  });
+		  auth0.crossOriginAuthenticationCallback();
+		</script>
+		</head>
+		<body></body>
+		</html>	  
+EOT;
```

Comment:
is this well closed? why dont use regular `"`?

Reply:
Really just style, for a single line I think " is best, for multiple lines I like heredoc.  http://php.net/manual/en/language.types.string.php#language.types.string.syntax.heredoc